src/Training.cpp

```diff
@@ -139,11 +139,25 @@ void Training::clear_training() {
     Training::m_data.clear();
 }
 
+TimeStep::NNPlanes Training::get_planes(const GameState* const state) {
+    auto input_data = std::vector<net_t>{};
+    Network::gather_features(state, input_data, 0);
```

Comment:
I guess gather_features could also just return the result. (Don't have to do this in this patch, I just noticed)

Reply:
I thought about that too but decided not to as this would involve data being copied from the stack on return. I'm happy to change it if you prefer it though.

Reply:
Look up Return Value Optimization.